subsys/net/lib/download_client/src/download_client.c

```diff
@@ -403,6 +407,62 @@ static int reconnect(struct download_client *dl)
 	return 0;
 }
 
+static size_t recv_block(struct download_client *dl)
+{
+	int err, timeout = 0;
+
+	switch (dl->proto) {
+	case IPPROTO_TCP:
+	case IPPROTO_TLS_1_2:
+		timeout = CONFIG_DOWNLOAD_CLIENT_TCP_SOCK_TIMEO_MS;
+		break;
+	case IPPROTO_UDP:
+	case IPPROTO_DTLS_1_2:
+		if (IS_ENABLED(CONFIG_COAP)) {
+			timeout = CONFIG_DOWNLOAD_CLIENT_UDP_SOCK_TIMEO_MS;
+			break;
+		}
+	default:
+		LOG_ERR("unhandled proto");
+		return -1;
+	}
+
+	err = set_recv_socket_timeout(dl->fd, timeout);
+	if (err) {
+		return -1;
+	}
+
+	return recv(dl->fd, dl->buf + dl->offset, sizeof(dl->buf) - dl->offset, 0);
+}
+
+static int send_block_request(struct download_client *dl)
```

Comment:
I presume `block` here refers to blocking behaviour, not block transfer (CoAP)? It might just be me, but I'd find `recv_blocking()`/`send_blocking()` a little less confusing when reading.